src/github/handlers/repository-dispatch.ts

```diff
@@ -0,0 +1,117 @@
+import { GitHubContext } from "../github-context";
+import { dispatchWorkflow, getDefaultBranch } from "../utils/workflow-dispatch";
+import { Value } from "@sinclair/typebox/value";
+import { DelegatedComputeInputs, PluginChainState, expressionRegex, pluginOutputSchema } from "../types/plugin";
+import { PluginChain } from "../types/config";
+
+export async function repositoryDispatch(context: GitHubContext<"repository_dispatch">) {
+  console.log("Repository dispatch event received", context.payload.client_payload);
+
+  if (context.payload.action !== "return_data_to_ubiquibot_kernel") {
+    console.log("Skipping non-ubiquibot event");
+    return;
+  }
+
+  let pluginOutput;
+
+  try {
+    pluginOutput = Value.Decode(pluginOutputSchema, context.payload.client_payload);
+  } catch (error) {
+    console.error("Cannot decode plugin output", error);
+    throw error;
+  }
+  console.log("Plugin output", pluginOutput);
+
+  const state = await context.eventHandler.pluginChainState.get(pluginOutput.state_id);
+  if (!state) {
+    console.error("No state found for plugin chain");
+    return;
+  }
+
+  if (!("installation" in state.eventPayload) || state.eventPayload.installation?.id === undefined) {
+    console.error("No installation found");
+    return;
+  }
+
+  const currentPlugin = state.pluginChain[state.currentPlugin];
+  if (currentPlugin.plugin.owner !== context.payload.repository.owner.login || currentPlugin.plugin.repo !== context.payload.repository.name) {
+    console.error("Plugin chain state does not match payload");
+    return;
+  }
+  state.outputs[state.currentPlugin] = pluginOutput;
+  console.log("State", state);
+
+  const nextPlugin = state.pluginChain[state.currentPlugin + 1];
+  if (!nextPlugin) {
+    console.log("No more plugins to call");
+    await context.eventHandler.pluginChainState.put(pluginOutput.state_id, state);
+    return;
+  }
+  console.log("Dispatching next plugin", nextPlugin);
+
+  const token = await context.eventHandler.getToken(state.eventPayload.installation.id);
+  const ref = nextPlugin.plugin.ref ?? (await getDefaultBranch(context, nextPlugin.plugin.owner, nextPlugin.plugin.repo));
```

Comment:
`Expression produces a union type that is too complex to represent.` 
This function also gets called in another part of the code and in that case there's no error like this, so I'm not sure what's too complex...

Reply:
Consider type guards. I was struggling with this same error when dealing with all the webhook types. I even tried hacking tsc and maxing memory limits to no avail.